### PR TITLE
gh-92120: The docstring of enum.Enum is invalid in reST

### LINKED ISSUE #92120: The docstring of `enum.Enum` is invalid in reStructuredText since 3.11
**Bug report**

The docstring of `enum.Enum` is invalid in reStructuredText since 3.11.

```
$ python3.11                                                                                                ~/work/tmp/doc
Python 3.11.0a7+ (heads/main:ad5e852, May  2 2022, 01:40:38) [Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import enum
>>> print(enum.Enum.__doc__)
(snip)
    - attribute access::

    >>> Color.RED
    <Color.RED: 1>
(snip)
>>>
```

The "attribute access" line ends with a code-block marker "::". But no code-block follows.

As a result, Sphinx emits a warning for the subclasses of Enum as following:

```
/Users/tkomiya/work/tmp/doc/example.py:docstring of example.Transpose:7: WARNING: Literal block expected; none found.
```

refs: https://github.com/sphinx-doc/sphinx/issues/10410

Additionally, the bullet list is interrupted by code-blocks. IMO, these code-blocks should be indented to the item level of the bullet list.

**Your environment**

- CPython versions tested on: Python 3.11.0a7+ (heads/main:ad5e852, May  2 2022, 01:40:38)
- Operating system and architecture: macOS 10.15

### Linked PRs
* gh-92122

### PR DESCRIPTION
**Pull request by tk0miya**
Closes #92120

* Issue: gh-92120

### CODE REVIEW

**Comment by hugovk** [comment]
Thank you Takeshi-san!

Are similar fixes needed in the `EnumType` class?

https://github.com/python/cpython/blob/ad5e8520f3e117f45481513014548a39879d30d2/Lib/enum.py#L504

https://github.com/python/cpython/blob/ad5e8520f3e117f45481513014548a39879d30d2/Lib/enum.py#L558

Please could you include a NEWS file?

https://devguide.python.org/committing/#updating-news-and-what-s-new-in-python

**Comment by ethanfurman** [comment]
I am more interested in the docstrings being usable by `doctest` than being correct reStructured text.  Have you verified that `doctest` correctly executes the docstrings with your changes applied?

**Comment by tk0miya** [comment]
Thank you for your reviews! I also fixed EnumTypes class.

> Have you verified that doctest correctly executes the docstrings with your changes applied?

Yes. I confirmed it on my local. This change does not prevent the result of doctest. Of course, the number of cases are not changed. Additionally, doctest on GitHub Action is also passed.
refs: https://github.com/python/cpython/runs/6310616547?check_suite_focus=true_

File: Misc/NEWS.d/next/Library/2022-05-06-01-53-34.gh-issue-92122.96Lf2p.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix grammatical error of the doctring of enum.Enum
```
Comment (AA-Turner) [inline Misc/NEWS.d/next/Library/2022-05-06-01-53-34.gh-issue-92122.96Lf2p.rst]:
```suggestion
Fix reStructuredText syntax errors in docstrings in the :mod:`enum` module.
```

Comment (tk0miya) [reply Misc/NEWS.d/next/Library/2022-05-06-01-53-34.gh-issue-92122.96Lf2p.rst]: Thanks. Updated!

**Review by AA-Turner — APPROVED** [review]

**Review by ethanfurman — APPROVED** [review]

**Pull request merged**